### DISCUSSION TITLE
Dynamic collections

### DISCUSSION THREAD

**Discussion by timuric** [saleor/saleor#14414] [Ideas]
Adding products to collections requires manual maintenance and use of external tools to create dynamic rules to create and update collections. Alternatively we can use filters to dynamically query products and treat them as a collection, however use of filters has few drawbacks:

1. Filtered results can only be sorted programmatically
2. No explicit definition or semantic meaning of filter rules. For example “summer-sale” collection communicates its intention
3. Filtered results can be used as a set for boolean operations (in sales and vouchers)
4. Communicate product count filtered results contain

If above limitations are lifted we can replace collection primitive with a more productive and universal abstraction that can empower users in following ways:

• Create dynamic sets that are always up to date

• Reuse sets for expressive sales and discount rules

• Use sets as navigation starting point

• Use sets in BI reporting 

• Narrow down products using boolean operations

**Comment by maarcingebala** [comment]
RFC: https://github.com/saleor/saleor/issues/12882
